### DISCUSSION TITLE
Release Cycle/Roadmap/Milestones/EOL

### DISCUSSION THREAD

**Discussion by tentwofour** [sulu/sulu#5788] [Q&A] [answered]
Hi, 

I've used Sulu for a couple of personal projects so far and love it; I'm looking to use it on a larger project now. I see some milestones in the 2.x and 3.0 area, but no due date. Just wondering if there's a path you're following with certain releases. Does 2.x's EOL somewhat align to Symfony 4.4LTS? Or to PHP version EOL's? It seems like more the latter, as I see an issue about dropping PHP5.5 that was closed because Symfony was still supporting it at the time...

Apologize if it's somewhere obvious, I took a look through issues here, and across your main site, but couldn't find anything.

**Comment by niklasnatter** [comment, marked as answer]
Hey,
thanks for reaching out! We are working on the content to communicate this through our homepage at the moment. Until that page is ready and polished, I will try to clarify things here 🙂 

Release Cycle:

- **Patch versions**: There is no fixed schedule for patch versions. We try to release new versions timely after fixing critical (or annoying) bugs. Patch releases will mainly include improvements and bugfixes.
- **Minor versions**: We intend to release a new minor version **every 6 months**. Releases might be delayed to a certain extent to include features like compatibility with new Symfony version. For example, the release of version 2.1 was delayed until we reached compatibility with Symfony 5. Minor releases will include new features and minor refactorings.
- **Major versions**: We plan to release a new major version **every 2-3 years**. Major versions will remove deprecated features, bump dependencies and include major refactorings. 

Maintained Versions:

- The **last minor version of the previous major version** (at the moment 1.6) is maintained as LTS version and will receive fixes for security issues and critical bugs.
- The **last two minor versions of the current major version** (at the moment 2.3 and 2.4) are actively maintained and will receive bugfixes and improvements. 
- We do offer paid support for unmaintained versions.

PHP/Symfony Compatibility:

- We aspire to be compatible with new PHP and Symfony versions as soon as possible. 
- If possible without breaking backward compatibility, we try to establish compatibility in patch versions. If new versions force us to break backward compatibility, we will establish compatibility in the next minor version. 

Backward Compatibility:

- We take backward compatibility seriously and try to avoid backward compatibility breaks in patch versions and minor versions if possible. These versions will contain backward compatibility breaks only if they are needed for fixing security issues, critical bugs or compatibility with new PHP and Symfony versions.
- Major versions will contain backward compatibility breaks.
- Every backward compatibility break will be documented in the [`UPGRADE.md` of the `sulu/sulu` package](https://github.com/sulu/sulu/blob/2.x/UPGRADE.md).

Fee free to add a comment here if something is not clear or if you have additional questions!

**Reply by tentwofour** [reply]
Awesome, thanks for the detailed reply! I understand it takes time to determine and communicate all of that amongst other projects in the queue.

That aligned to most of my guesses, looking through the repository/issues/changelog - hope I can contribute as I get into using Sulu more.

**Reply by simontol** [reply]
Hi, I've read that version 3.0 is scheduled for release this year. Is there an indicative date when version 1.6 will become unmaintained? We have a big project running on version 1.6 and we should plan to upgrade. 

Thanks, Simone.

**Reply by niklasnatter** [reply]
Hey, thanks for reaching out. Yes you are right, we plan to finish and release Sulu 3.0 at the end of 2022. With the release of Sulu 3.0, Sulu 1.6 will reach its end of life and will no longer be maintained. 

We will still offer paid support for unmaintained versions if necessary, but in general I would definitely recommend upgrading your project 🙂
